### PR TITLE
[bitnami/harbor] Add updateStrategy, fix replicas

### PR DESCRIPTION
**Pull request by funkypenguin**
**Description of the change**

Unlike the [official helm chart](https://github.com/goharbor/harbor-helm), this chart doesn't currently allow the user to set an updateStrategy for the deployments (chartmuseum, registry, jobservice) which would usually be attached to storage in RWO mode. In this event, an update (i.e., via `helm upgrade`) will fail, since the _old_ instance of a pod will not detach from its storage, preventing the _new_ instance of a pod from replacing it.

This change ports the updateStrategy.type value from the official helm chart for the chartmuseum, jobservice, and registry deployments. 

The PR additionally fixes an error re the replicas for notary service/signer, which failed local CI when preparing this PR.

**Benefits**

Users can choose (non-default) to set the updateStrategy to "Recreate", so that helm updates on RWO/1-replica deployments succeed.

**Possible drawbacks**

None. The default behaviour has not changed.

**Applicable issues**

  - fixes #

**Additional information**

**Checklist** 
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)

:warning: Keep in mind that if you want to make changes to the kubeapps chart, please implement them in the [kubeapps repository](https://github.com/kubeapps/kubeapps/tree/master/chart/kubeapps). This is only a synchronized mirror.

### CODE REVIEW

File: bitnami/harbor/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: harbor
-version: 3.1.4
```
Comment (miguelaeh) [inline bitnami/harbor/Chart.yaml]: Please bump the minor version.

File: bitnami/harbor/requirements.lock [hunk 2]
```diff
@@ -1,9 +1,9 @@
 dependencies:
 - name: postgresql
   repository: https://charts.bitnami.com/bitnami
-  version: 7.7.3
+  version: 8.x.x
```
Comment (miguelaeh) [inline bitnami/harbor/requirements.lock]: I think doing this the version should be a major.
Also, we should test these changes properly in our pipeline so please, open a different PR for the dependencies update.

File: bitnami/harbor/values.yaml [hunk 3]
```diff
@@ -805,6 +807,14 @@ core:
 ##
 jobservice:
   replicas: 1
+
+  ## Update strategy - only really applicable for deployments with RWO PVs attached
+  ## If replicas = 1, an update can get "stuck", as the previous pod remains attached to the 
+  ## PV, and the "incoming" pod can never start. Changing the strategy to "Recreate" will 
+  ## terminate the single previous pod, so that the new, incoming pod can attach to the PV
+  updateStrategy:
```
Comment (miguelaeh) [inline bitnami/harbor/values.yaml]: Add a line after the comments:
```diff
## Update strategy - only really applicable for deployments with RWO PVs attached
## If replicas = 1, an update can get "stuck", as the previous pod remains attached to the 
## PV, and the "incoming" pod can never start. Changing the strategy to "Recreate" will 
## terminate the single previous pod, so that the new, incoming pod can attach to the PV
+ ##
```

File: bitnami/harbor/values.yaml [hunk 4]
```diff
@@ -869,6 +879,14 @@ jobservice:
 
 registry:
   replicas: 1
+
+  ## Update strategy - only really applicable for deployments with RWO PVs attached
+  ## If replicas = 1, an update can get "stuck", as the previous pod remains attached to the 
+  ## PV, and the "incoming" pod can never start. Changing the strategy to "Recreate" will 
+  ## terminate the single previous pod, so that the new, incoming pod can attach to the PV
+  updateStrategy:
```
Comment (miguelaeh) [inline bitnami/harbor/values.yaml]: Same as the previous one.

File: bitnami/harbor/values.yaml [hunk 5]
```diff
@@ -971,6 +989,14 @@ registry:
 chartmuseum:
   enabled: true
   replicas: 1
+
+  ## Update strategy - only really applicable for deployments with RWO PVs attached
+  ## If replicas = 1, an update can get "stuck", as the previous pod remains attached to the 
+  ## PV, and the "incoming" pod can never start. Changing the strategy to "Recreate" will 
+  ## terminate the single previous pod, so that the new, incoming pod can attach to the PV
+  updateStrategy:
```
Comment (miguelaeh) [inline bitnami/harbor/values.yaml]: Same as the previous one.

File: bitnami/harbor/values.yaml [hunk 6]
```diff
@@ -1142,8 +1168,9 @@ clair:
 ##
 notary:
   enabled: true
-  replicas: 1
```
Comment (miguelaeh) [inline bitnami/harbor/values.yaml]: Why did you remove this?

File: bitnami/harbor/templates/clair/clair-dpl.yaml [hunk 7]
```diff
@@ -6,7 +6,7 @@ metadata:
   labels: {{- include "harbor.labels" . | nindent 4 }}
     app.kubernetes.io/component: clair
 spec:
-  replicas: {{ .Values.clair.replicas }}
+  replicas: {{ .Values.clair.replicas }}
```
Comment (miguelaeh) [inline bitnami/harbor/templates/clair/clair-dpl.yaml]: Delete the space at the final, this line is not actually changed.

File: bitnami/harbor/requirements.lock (hunk 2 above)
Comment (juan131) [reply bitnami/harbor/requirements.lock]: I already did that at https://github.com/bitnami/charts/pull/1936
Please rebase from master instead.

**Comment by funkypenguin** [comment]
Oooer, I've messed up rebase. Standby...

**Comment by funkypenguin** [comment]
Sorry chaps, I can't see what I did wrong when executing the rebase (I'm a rebasing noob). My branch now matches the latest master, but the PR lists a huge amount of files changed at https://github.com/bitnami/charts/pull/1926/files - what did I do wrong?

D

**Comment by juan131** [comment]
Hi @funkypenguin 

If you're struggling with the rebase, I recommend you to create a complete new PR (closing this one).

**Comment by funkypenguin** [comment]
Agreed, I think that'll be easier. Who know what kind of mess I've made... :)

**Comment by funkypenguin** [comment]
For reference, the replacement PR is https://github.com/bitnami/charts/pull/1953
